packages/runtime/src/state/actions.ts

```diff
@@ -239,12 +239,12 @@ type MessageBuilderPropControllerAction<T = PropControllerMessage> = {
 
 type ChangeAPIResourceAction = {
   type: typeof ActionTypes.CHANGE_API_RESOURCE
-  payload: { resource: APIResource }
+  payload: { resource: APIResource; locale: string | null }
```

Comment:
Added an optional `locale` argument to actions operating on the API resources state.

Reply:
Should `locale` here be an optional field? In particular, is this action ever constructed by the builder (it's unfortunate I even have to ask this question but we are reusing these actions there... ideally we'd have a `Builder` and `Host` interface that encapsulates the parts of the builder and host that talk to each other, but I digress)?

Reply:
In particular, when we handle the action payload in a reducer, the action received from the builder might not include the field at all so it wouldn't be neither `string` or `null`, it would be `undefined`

Reply:
I have `locale` as optional in the action(s) "constructor(s)", which converts `undefined` to `null`: https://github.com/makeswift/makeswift/pull/839/files#diff-f587b032a852ddc3abcdb9915e263b0a3b74536e71fc024e97e0a13623858cb9R658

Reply:
Right, but the current builder dependency doesn't import that action creator so it will send `undefined` until the dependency is upgraded. If we parsed actions at runtime with Zod, this action would fail to parse if a host used this version of the runtime but the builder wasn't updated yet.

Reply:
Ah, I see, good point.